File: apps/macos/PasteSheets/Domain/UseCases/ManageDirectoriesUseCase.swift
import Foundation

final class ManageDirectoriesUseCase {
    private let repo: DirectoryRepository

    init(repo: DirectoryRepository) {
        self.repo = repo
    }

    func getAllDirectories() throws -> [DirectoryInfo] {
        try repo.getAllDirectories()
    }

    func createDirectory(name: String) throws -> Int64 {
        try repo.createDirectory(name: name)
    }

    func renameDirectory(oldName: String, newName: String) throws {
        try repo.renameDirectory(oldName: oldName, newName: newName)
    }

    func deleteDirectory(name: String) throws {
        try repo.deleteDirectory(name: name)
    }
}

File: apps/windows/PasteSheet/Domain/UseCases/ManageDirectoriesUseCase.cs
using PasteSheet.Domain.Entities;
using PasteSheet.Domain.Repositories;

namespace PasteSheet.Domain.UseCases;

public sealed class ManageDirectoriesUseCase
{
    private readonly IDirectoryRepository _repo;

    public ManageDirectoriesUseCase(IDirectoryRepository repo) => _repo = repo;

    public List<DirectoryInfo> GetAllDirectories() => _repo.GetAllDirectories();

    public long CreateDirectory(string name) => _repo.CreateDirectory(name);

    public void RenameDirectory(string oldName, string newName) =>
        _repo.RenameDirectory(oldName, newName);

    public void DeleteDirectory(string name) => _repo.DeleteDirectory(name);
}
